src/Bundle/Maker/Factory/MakeFactoryQuery.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Zenstruck\Foundry\Bundle\Maker\Factory;
+
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Component\Console\Input\InputInterface;
+
+/** @internal */
+final class MakeFactoryQuery
+{
+    private function __construct(private string $namespace, private bool $test, private bool $persisted, private bool $allFields, private string $class, private Generator $generator)
+    {
+    }
+
+    public static function fromInput(InputInterface $input, Generator $generator): self
+    {
+        return new self(
+            namespace: $input->getOption('namespace'),
+            test: (bool) $input->getOption('test'),
+            persisted: !$input->getOption('no-persistence'),
+            allFields: (bool) $input->getOption('all-fields'),
+            class: $input->getArgument('class'),
+            generator: $generator
+        );
+    }
+
+    public function getNamespace(): string
+    {
+        return $this->namespace;
+    }
+
+    public function isTest(): bool
+    {
+        return $this->test;
+    }
+
+    public function isPersisted(): bool
+    {
+        return $this->persisted;
+    }
+
+    public function isAllFields(): bool
+    {
+        return $this->allFields;
+    }
+
+    public function getClass(): string
+    {
+        return $this->class;
+    }
+
+    public function getGenerator(): Generator
```

Comment:
I'm not 100% happy with the generator in this class, I think I'll take it out, but it was easier when I created the feature